src/controllers/MainController.js

```diff
@@ -3,30 +3,64 @@ class MainController {
     this.service = serviceName;
   }
 
-  async get({ params: { id } = {} }, res) {
-    const response = await this.service.get(id);
-    res.json(response);
-    res.sendStatus(response ? 200 : 404);
+  async get({ params: { id } = {} }, res, next) {
+    try {
+      const response = await this.service.get(id);
+      if (!response) {
+        const err = new Error('Not found');
+        err.status = 404;
+        throw err;
```

Comment:
```suggestion
        throw err.notFound;
```

You use this part of code in several places. Try to keep DRY principle